source/Octopus.Client/Repositories/Async/MixedScopeBaseRepository.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Collections.Generic;
+using Octopus.Client.Exceptions;
+using Octopus.Client.Extensibility;
+using Octopus.Client.Util;
+
+namespace Octopus.Client.Repositories.Async
+{
+    class MixedScopeBaseRepository<TMixedScopeResource>: BasicRepository<TMixedScopeResource> where TMixedScopeResource : class, IResource
+    {
+        public MixedScopeBaseRepository(IOctopusAsyncClient client, string collectionLinkName) : base(client, collectionLinkName)
+        {
+        }
+
+        SpaceQueryParameters SpaceQueryParameters { get; set; }
+        protected void SetupParameters(bool includeGlobal, string[] spaceIds)
```

Comment:
Can we pass this in as a constructor parameter instead of modifying an already constructed repo?